src/context/authStore/bucketStorageHelper.ts

```diff
@@ -0,0 +1,129 @@
+ * data and will be stored in the first bucket instead of deleting it,
+ * differentiating an empty string from no data.
+ *
+ * Note that the function does not read the existing value stored and takes
+ * reference from {@link oldValue} instead.
+ *
+ * All storage calls are done asynchronously without blocking the caller.
+ *
+ * @param newValue new value to be stored
+ * @param oldValue previous value stored
+ * @param bucketSize number of keys per bucket
+ * @param storageKey key to be used for storing
+ */
+export const saveToStoreInBuckets: (
+  storageKey: string,
+  newValue: string,
+  oldValue: string | null
+) => void = (storageKey, newValue, oldValue) => {
+  oldValue = oldValue ?? "";
+
+  // iterate the longer length in case we have to clear buckets
+  const compareLength = Math.max(newValue.length, oldValue.length);
+  for (let bucketNo = 0; bucketNo * SIZE_LIMIT < compareLength; bucketNo++) {
+    const newValueBucketString = newValue.slice(
+      bucketNo * SIZE_LIMIT,
+      (bucketNo + 1) * SIZE_LIMIT
+    );
+    const oldValueBucketString = oldValue.slice(
+      bucketNo * SIZE_LIMIT,
+      (bucketNo + 1) * SIZE_LIMIT
+    );
+
+    if (newValueBucketString !== oldValueBucketString) {
+      if (newValueBucketString === "" && bucketNo !== 0) {
+        // if the new bucket is empty, delete the key
+        // however, if this is the first bucket, user is saving an empty string, so do not delete
+        SecureStore.deleteItemAsync(getBucketKey(storageKey, bucketNo));
+      } else {
+        SecureStore.setItemAsync(
+          getBucketKey(storageKey, bucketNo),
+          newValueBucketString
+        );
+      }
+    }
+  }
+};
+
+/**
+ * Reads data from storage in buckets and returns it. If no data was read, returns null.
+ * An empty string will be counted as data. This differentiates it from no data.
+ *
+ * @param storageKey key used in storage
+ * @returns the value of the data in storage, null if there was no data found
+ */
+export const readFromStoreInBuckets: (
+  storageKey: string
+) => Promise<string | null> = async (storageKey) => {
+  let bucketNo = 0;
+  let storageData = "";
+  while (true) {
+    const newValueBucketString = await SecureStore.getItemAsync(
+      getBucketKey(storageKey, bucketNo)
+    );
+    if (newValueBucketString === undefined) {
+      // this shouldn't happen unless SecureStore is mocked with no implementation
+      throw new Error(
+        "unexpected undefined here, you probably should mock SecureStore"
+      );
+    }
+
+    if (newValueBucketString === null) {
+      break;
+    }
+    storageData += newValueBucketString;
+    bucketNo++;
+  }
+  return bucketNo === 0 ? null : storageData;
+};
+
+/**
+ * Deletes all buckets from storage. The number of existing buckets is determined by
+ * the length of {@link oldValue}.
+ *
+ * @param storageKey key used in storage
+ * @param oldValue existing value in storage
+ */
+export const deleteInStoreInBuckets: (
```

Comment:
Just noticed that the naming is a little confusing, would `deleteStoreInBuckets` suffice?

Reply:
Also, do we need this function? Noticed it isn't being used outside of tests

Reply:
not currently being used in the authStore but just added for completeness of the utility functions